src/Common/src/TypeSystem/Common/TypeDesc.cs

```diff
@@ -99,7 +99,7 @@ public abstract partial class TypeDesc
         public override int GetHashCode()
         {
             // Inherited types are expected to override
-            return RuntimeHelpers.GetHashCode(this);
+            throw new NotImplementedException();
```

Comment:
This doesn't look right. This MSDN article https://msdn.microsoft.com/en-us/library/ms173150(VS.80).aspx says otherwise and I just tried the following which compiles just fine:
```C#
public class D {
    public override int GetHashCode() {
		return 1;
    }
}

public abstract class E : D {
    public abstract override int GetHashCode ();
}

public class F : E {
    public override int GetHashCode() {
		return 2;
    }
}
```